app/protocol/flowcontext/ibd.go

```diff
@@ -36,29 +42,42 @@ func (f *FlowContext) IsInIBD() bool {
 
 // selectPeerForIBD returns the first peer whose selected tip
 // hash is not in our DAG
-func (f *FlowContext) selectPeerForIBD(dag *blockdag.BlockDAG) *peerpkg.Peer {
+func (f *FlowContext) selectPeerForIBD() (*peerpkg.Peer, error) {
 	f.peersMutex.RLock()
 	defer f.peersMutex.RUnlock()
 
 	for _, peer := range f.peers {
 		peerSelectedTipHash := peer.SelectedTipHash()
-		if !dag.IsInDAG(peerSelectedTipHash) {
-			return peer
+		blockInfo, err := f.domain.Consensus().GetBlockInfo(peerSelectedTipHash)
+		if err != nil {
+			return nil, err
+		}
+		if !blockInfo.Exists {
+			return peer, nil
 		}
 	}
-	return nil
+	return nil, nil
 }
 
 func (f *FlowContext) requestSelectedTipsIfRequired() {
-	if f.isDAGTimeCurrent() {
+	dagTimeCurrent, err := f.isDAGTimeCurrent()
+	if err != nil {
+		panic(err)
+	}
+	if dagTimeCurrent {
 		return
 	}
 	f.requestSelectedTips()
 }
 
-func (f *FlowContext) isDAGTimeCurrent() bool {
+func (f *FlowContext) isDAGTimeCurrent() (bool, error) {
```

Comment:
Please rename this method to shouldRequestSelectedTips to avoid confusion with the similar functionality inside consensus.